### PR TITLE
Merge or remove useless cases from bugbuster to ICG

### PR DESCRIPTION
**Pull request by pengzhout**
Signed-off-by: Haozhou Wang <hawang@pivotal.io>

### CODE REVIEW

File: src/test/regress/expected/arrays.out [hunk 1]
```diff
@@ -1318,3 +1318,8 @@ NOTICE:  removing built-in function "int_agg_state"
 RESET SESSION AUTHORIZATION;
 DROP USER IF EXISTS user_internal_stype;
 -- end_ignore
+select unnest('{}'::text[]);
```
Comment (hlinnaka) [inline src/test/regress/expected/arrays.out]: 'arrays' is an upstream file. Let's try to keep it unchanged, to make merging easier.

**Comment by danielgustafsson** [comment]
`src/test/regress/parallel_schedule` is also inherited from upstream, can we add the testcases to `greenplum_schedule` instead?

**Comment by pengzhout** [comment]
@danielgustafsson thanks, fixed.

**Comment by danielgustafsson** [comment]
In `gp_persistent_table`, what is the point of this test?:
```SQL
SET gp_enable_inheritance = true;
ERROR:  unrecognized configuration parameter "gp_enable_inheritance"
```
bugbuster tests are the only references to this GUC, is it an historical artefact that we can remove?

**Comment by hlinnaka** [comment]
What's the point of the 'world' test? It seems rather uninteresting, I think it's just setting up things in preparation for other tests in the bugbuster suite, rather than testing anything useful on its own. Also, we seem to have the exact same tables in in the 'qp_with_clause' test already.

**Comment by hlinnaka** [comment]
TestPSQL is in need of some serious cleanup.

Firstly, it's completely enclosed in a start/end_ignore block, so it doesn't test that the commands return what we expect at all. At best, it will catch if one of the commands make the system crash. So that makes many of the tests in the file rather pointless.

Secondly, what's up with the "drop database" commands at the beginning of TestPSQL.sql? Seems completely unrelated to anything else in the file.

Thirdly, I find these rather scary:

update gp_segment_configuration set role='p' where role='t' and content=0;
ERROR:  permission denied: "gp_segment_configuration" is a system catalog

If that actually succeeded, that might cause some pretty surprising issues later in the regression run, which could be really hard to debug.

**Comment by asimrp** [comment]
On Wed, Nov 9, 2016 at 1:22 AM, Daniel Gustafsson <notifications@github.com>
wrote:

> In gp_persistent_table, what is the point of this test?:
>
> SET gp_enable_inheritance = true;
> ERROR:  unrecognized configuration parameter "gp_enable_inheritance"
>
> bugbuster tests are the only references to this GUC, is it an historical
> artefact that we can remove?
>

Good catch, let's remove it please.

**Comment by pengzhout** [comment]
@danielgustafsson @hlinnaka  good catch, fixed them

**Comment by pengzhout** [comment]
pushed with comments fixed.

**Pull request closed without merging**